primitives/fee-handler/src/mock.rs

```diff
@@ -0,0 +1,210 @@
+// Ajuna Node
+// Copyright (C) 2022 BlogaTech AG
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::{
+	fee_handler::{AssetGameFeeHandler, DistributeFee, Payment},
+	withdraw_credit::{EnsureWhitelistedAsset, WithdrawWhitelistedCredit},
+	NativeGameFeeHandler, WithdrawAsset, WithdrawNative,
+};
+use frame_support::{
+	derive_impl,
+	traits::{AsEnsureOriginWithArg, ConstU32},
+	BoundedVec,
+};
+use sp_runtime::{
+	testing::TestSignature,
+	traits::{IdentifyAccount, Verify},
+	BuildStorage, DispatchError, TokenError,
+};
+
+pub type Signature = TestSignature;
+pub type AccountSignature = <Signature as Verify>::Signer;
+pub type AccountId = <AccountSignature as IdentifyAccount>::AccountId;
+pub type Block = frame_system::mocking::MockBlock<Test>;
+pub type Balance = u64;
+pub type AssetId = u32;
+
+pub const ALICE: AccountId = 1;
+pub const BOB: AccountId = 2;
+pub const CHARLIE: AccountId = 3;
+pub const DAVE: AccountId = 4;
+pub const FERDIE: AccountId = 5;
+
+pub const TOURNAMENT_TREASURY: AccountId = 431;
+
+pub const WHITELISTED_ASSET_ID: AssetId = 888;
+pub const NOT_WHITE_LISTED_ASSET_ID: AssetId = 999;
+
+// Configure a mock runtime to test the pallet.
+frame_support::construct_runtime!(
+	pub struct Test {
+		System: frame_system = 0,
+		Balances: pallet_balances = 1,
+		Assets: pallet_assets = 2,
+	}
+);
+
+#[derive_impl(frame_system::config_preludes::TestDefaultConfig)]
```

Comment:
This is actually very cool, and we might want to use it in other cases too in order to make our mocks less verbose.